### PR TITLE
Add parameter support for the ROS and ROS2 cpp code generator

### PR DESCRIPTION
**Pull request by ipa-hsd**
Addresses https://github.com/ipa320/ros-model/issues/75

### CODE REVIEW

**Comment by ipa-hsd** [comment]
For ros model
```
Parameters {
    Parameter { name 'string_test' type String default 'hello'},
    Parameter { name 'double_test' type Double default 1.0}}
```
the code generated is
```
this->declare_parameter<std::string>("string_test", "hello");
this->declare_parameter<double>("double_test", 1.0);
```

File: plugins/de.fraunhofer.ipa.roscode.generator/src/de/fraunhofer/ipa/roscode/generator/Ros2CodeGenerator.xtend [hunk 1]
```diff
@@ -248,4 +280,68 @@ int main(int argc, char * argv[])
...
+  		} else {
+	  		var param_pair = elem.getType().get_param_type();
+  			struct_str += "this->declare_parameter<" + param_pair.getKey() + ">(\"" + elem.getName() + param_pair.getValue() + "\");\n";
+//  			struct_str += "this->get_parameter(\"" + elem.getName() + "\", " + elem.getName() + "_);\n\n";
+  		}
+  	}
+	return struct_str;
+ }
+
+ def Pair<String, String> get_param_type(ParameterType type) {
```
Comment (ipa-nhg) [inline plugins/de.fraunhofer.ipa.roscode.generator/src/de/fraunhofer/ipa/roscode/generator/Ros2CodeGenerator.xtend:309]: is it possible to move "get_param_type" to another class, called for example "code_generator_helper" that can be used by the ROS2 and the ROS1 code generator?

Similar to https://github.com/ipa320/ros-model/blob/master/plugins/de.fraunhofer.ipa.rossystem.xtext/src/de/fraunhofer/ipa/rossystem/generator/GeneratorHelpers.xtend

**Comment by ipa-nhg** [comment]
@ipa-hsd thanks a lot! It works to me now! 👍🏽 

To complete the PR we have to fix the indentations of the new added lines and update the tests, see: https://github.com/ipa320/ros-model/pull/136/checks?check_run_id=2710375384#step:3:13806

File: plugins/de.fraunhofer.ipa.roscode.generator/src/de/fraunhofer/ipa/roscode/generator/RosCodeGenerator.xtend [hunk 2]
```diff
@@ -141,36 +152,46 @@ void  «sub.name»_cb (const «sub.message.package.name»::«sub.message.name»
...
+	ros::init(argc, argv, "«node.name»");
+  	ros::NodeHandle n;
+  	«FOR param : node.parameter»
+  	«IF (param.type instanceof ParameterStructTypeImpl)»
+  	«parameter_helper.compile_struct(param.type as ParameterStructTypeImpl, param.name).getKey()»
+  	«ELSE»
+  	«var param_pair = parameter_helper.get_param_type(param.type)»
+  	«IF !(param_pair.getKey().empty)»
+  	«param_pair.getKey()» «param.name»_;
+  	n.param("«param.name»", «param.name»_«param_pair.getValue()»);
```
Comment (ipa-nhg) [inline plugins/de.fraunhofer.ipa.roscode.generator/src/de/fraunhofer/ipa/roscode/generator/RosCodeGenerator.xtend:164]: a " is missed at the end.

It should be 

> n.param("«param.name»", «param.name»_«param_pair.getValue()»**"**);

Comment (ipa-hsd) [reply plugins/de.fraunhofer.ipa.roscode.generator/src/de/fraunhofer/ipa/roscode/generator/RosCodeGenerator.xtend:164]: The problem is here actually: https://github.com/ipa320/ros-model/blob/5886dc0a821e65cb4a95d424898eb1bf602127d4/plugins/de.fraunhofer.ipa.roscode.generator/src/de/fraunhofer/ipa/roscode/generator/RosCodeGenerator.xtend#L54

**Pull request merged**